metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala

```diff
@@ -134,24 +138,33 @@ class SemanticdbTreePrinter(
       case s.StringConstant(value) => value
     }
 
-  def printTree(t: s.Tree): Option[String] =
+  def printTree(
+      t: s.Tree,
+      isExplicitTuple: => Boolean = false
+  ): Option[String] =
     t match {
       case s.Tree.Empty => None
       case s.OriginalTree(_) => None
-      case s.TypeApplyTree(function, typeArguments) =>
-        Some(printTree(function).getOrElse("") + printTypeArgs(typeArguments))
+      case s.TypeApplyTree(function, typeArguments)
+          if shouldPrintTypeApply(function) || isExplicitTuple =>
+        Some(
+          printTree(function)
+            .filterNot(nonPrintable)
+            .getOrElse("") + printTypeArgs(typeArguments)
+        )
       case s.ApplyTree(function, arguments) =>
         Some(printTree(function).getOrElse("") + printArgs(arguments))
       case s.LiteralTree(constant) =>
         Some(printConstant(constant))
       case s.SelectTree(_, id) =>
-        id.flatMap(printTree)
+        id.flatMap(printTree(_))
```

Comment:
```suggestion
        id.flatMap(printTree)
```

Reply:
I'm pretty sure the `(_)` is needed now that there is a second default param in `printTree`.

Reply:
oh, you right!